crates/turborepo-lib/src/package_changes_watcher.rs

```diff
@@ -167,121 +228,133 @@ impl Subscriber {
-                    Ok(Ok(Event { paths, .. })) => {
-                        // No point in raising an error for an invalid .gitignore
-                        // This is slightly incorrect because we should also search for the
-                        // .gitignore files in the workspaces.
-                        let (root_gitignore, _) =
-                            Gitignore::new(&self.repo_root.join_component(".gitignore"));
-
-                        let changed_files: HashSet<_> = paths
-                            .into_iter()
-                            .filter_map(|p| {
-                                let p = AbsoluteSystemPathBuf::try_from(p).ok()?;
-                                self.repo_root.anchor(p).ok()
-                            })
-                            .filter(|p| {
-                                // If in .gitignore or in .git, filter out
-                                !(ancestors_is_ignored(&root_gitignore, p) || is_in_git_folder(p))
-                            })
-                            .collect();
-
-                        let changes = change_mapper.changed_packages(changed_files.clone(), None);
-
-                        match changes {
-                            Ok(PackageChanges::All) => {
-                                // We tell the client that we need to rediscover the packages, i.e.
-                                // all bets are off, just re-run everything
-                                let _ = self
-                                    .package_change_events_tx
-                                    .send(PackageChangeEvent::Rediscover);
-                                match self.initialize_repo_state().await {
-                                    Some(new_repo_state) => {
-                                        repo_state = new_repo_state;
-                                        change_mapper = match repo_state.get_change_mapper() {
-                                            Some(change_mapper) => change_mapper,
-                                            None => {
-                                                break;
-                                            }
-                                        };
-                                    }
-                                    None => {
-                                        break;
-                                    }
-                                }
-                            }
-                            Ok(PackageChanges::Some(mut changed_pkgs)) => {
-                                if !changed_pkgs.is_empty() {
-                                    tracing::debug!(
-                                        "changed files: {:?} changed packages: {:?}",
-                                        changed_files,
-                                        changed_pkgs
-                                    );
-                                }
+                interval.tick().await;
```

Comment:
100ms should be enough time that we don't need to park/notify, right?

Reply:
Yeah, I think that should be ok. We can ITG once the feature is stable.